src/ValueCalculator.java

```diff
@@ -0,0 +1,55 @@
+import java.util.Arrays;
+
+public class ValueCalculator {
+    private double[] realNumbers;
+    private int arraySize;
+    private int arrayHalfSize;
+
+    public ValueCalculator(int arraySize) {
+        if (arraySize < 1_000_000) {
+            arraySize = 1_000_000;
+        }
+        this.arraySize = arraySize;
+        this.arrayHalfSize = arraySize / 2;
+        this.realNumbers = new double[arraySize];
+    }
+
+    public int getArraySize() {
+        return arraySize;
+    }
+
+    public int getArrayHalfSize() {
+        return arrayHalfSize;
+    }
+
+    public void calculator() {
+        long start = System.currentTimeMillis();
+        Arrays.fill(realNumbers, 1);
+        double[] array1 = Arrays.copyOfRange(realNumbers, 0, arrayHalfSize);
+        double[] array2 = Arrays.copyOfRange(realNumbers, arrayHalfSize, arraySize);
+        Thread thread1 = new Thread(() -> {
+            for (int i = 0; i < array1.length; i++) {
+                array1[i] = (float) (array1[i] * Math.sin(0.2f + i / 5) * Math.cos(0.2f + i / 5) * Math.cos(0.4f + i / 2));
+            }
```

Comment:
this logic is duplicated, extract it into a method or create a class that implements runnable and move it there